torch/csrc/jit/passes/batch_mm.cpp

```diff
@@ -0,0 +1,173 @@
+#include "torch/csrc/jit/passes/batch_mm.h"
+
+#include "torch/csrc/jit/passes/dead_code_elimination.h"
+#include "torch/csrc/jit/interned_strings.h"
+#include "torch/csrc/utils/functional.h"
+
+#include <ATen/ATen.h>
+#include <algorithm>
+#include <unordered_map>
+
+namespace torch { namespace jit {
+
+// This pass looks for trees in the graph, where leaves are mm ops, and the inner
+// vertices are add nodes. Once we have such a tree they can be reduced to two
+// concats and a single mm (basically into a single multiply of a wide matrix, with
+// a tall matrix).
```

Comment:
Can you write the math for a simple case here? E.g., AB + CD == ...something...

Reply:
yup